### PR TITLE
cmake: Remove 32 bit support for VkLayer_khronos_shader_object

### PR DESCRIPTION
**Pull request by juan-lunarg**

### CODE REVIEW

File: layers/CMakeLists.txt [hunk 1]
```diff
@@ -15,11 +15,15 @@
...
 # limitations under the License.
 # ~~~
+set(TARGET_NAMES VkLayer_khronos_timeline_semaphore VkLayer_khronos_synchronization2 VkLayer_khronos_shader_object)
 
-if(WIN32)
-    set(TARGET_NAMES VkLayer_khronos_synchronization2 VkLayer_khronos_shader_object) # timeline_semaphore doesn't build on windows yet
-else()
-    set(TARGET_NAMES VkLayer_khronos_timeline_semaphore VkLayer_khronos_synchronization2 VkLayer_khronos_shader_object)
+# timeline_semaphore doesn't build on windows yet
+if (NOT WIN32)
```
Comment (lunarpapillo) [inline layers/CMakeLists.txt:21]: Isn't this backwards?  should be `if(WIN32)`, not `if(NOT WIN32)`, right?

File: layers/CMakeLists.txt [hunk 2]
```diff
@@ -15,11 +15,15 @@
...
-if(WIN32)
-    set(TARGET_NAMES VkLayer_khronos_synchronization2 VkLayer_khronos_shader_object) # timeline_semaphore doesn't build on windows yet
-else()
-    set(TARGET_NAMES VkLayer_khronos_timeline_semaphore VkLayer_khronos_synchronization2 VkLayer_khronos_shader_object)
+# timeline_semaphore doesn't build on windows yet
+if (WIN32)
+    list(REMOVE_ITEM ${TARGET_NAMES} "VkLayer_khronos_timeline_semaphore")
+endif()
+
+if (CMAKE_SIZEOF_VOID_P EQUAL "4")
```
Comment (lunarpapillo) [inline layers/CMakeLists.txt:25]: This is unusual, so a comment saying that the Vulkan SDK has deliberately chosen not to ship 32-bit `VkLayer_khronos_shader_object` files would save confusion in the future.

**Review by lunarpapillo — CHANGES_REQUESTED** [review]
You caught the logic inversion on `if (NOT WIN32)` before I could report it. :-)

One comment addition, then consider it approved by me.

**Comment by juan-lunarg** [comment]
64 bit Windows installation:
```
-- Installing: D:/a/Vulkan-ExtensionLayer/Vulkan-ExtensionLayer/build/./install/lib/VkLayer_khronos_synchronization2.json
-- Installing: D:/a/Vulkan-ExtensionLayer/Vulkan-ExtensionLayer/build/./install/lib/VkLayer_khronos_synchronization2.dll
-- Installing: D:/a/Vulkan-ExtensionLayer/Vulkan-ExtensionLayer/build/./install/lib/VkLayer_khronos_shader_object.json
-- Installing: D:/a/Vulkan-ExtensionLayer/Vulkan-ExtensionLayer/build/./install/lib/VkLayer_khronos_shader_object.dll
```

32 bit Windows installation:
```
-- Installing: D:/a/Vulkan-ExtensionLayer/Vulkan-ExtensionLayer/build/./install/lib/VkLayer_khronos_synchronization2.json
-- Installing: D:/a/Vulkan-ExtensionLayer/Vulkan-ExtensionLayer/build/./install/lib/VkLayer_khronos_synchronization2.dll
```

@lunarpapillo I think this is good to go now.

**Comment by jeremyg-lunarg** [comment]
@juan-lunarg do you still want this PR?

**Comment by juan-lunarg** [comment]
Yes we still need this PR for the Windows SDK release

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

**Pull request merged**